src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java

```diff
@@ -297,7 +298,9 @@ else if (replication != null)
                        .forEach(dc -> options.putIfAbsent(dc, defaultReplicas.toParseableString()));
         }
 
-        options.values().removeAll(Collections.singleton("0"));
+        // We do not filter out "0" due to the AUTH_KEYSPACE validation in validateExpectedOptions.
+        // The previousOptions are unavailable to that check so it is not possible to verify that all valid
+        // datacenters have an explicit configuration (even if it is 0).
```

Comment:
I don't think this is needed in general, just some specific operational checks that should be changed.

Reply:
I think the old code to filter out "0" should be retained, and the comment is no longer needed if you keep it.